### PR TITLE
Remove dynamically settable length of hash

### PR DESCRIPTION
**Pull request by Mad-Kat**
- Replaced `sha3_256` with `xxHash32` as it produces a smaller output (32bit vs 256bit)
- Replaced Base64 with Base62 as it generates [a-zA-Z0-9] and removed the obsolete replacement of certain characters

### CODE REVIEW

**Comment by jantimon** [comment]
does this pr increase the likelihood of collisions?

**Comment by Mad-Kat** [comment]
I think it decreases it.
It's not easy to exactly say it but we can try to estimate:

- Before: 5 characters in Base64 (5ˆ64) ~ 5.4*10ˆ44
- After: 6 characters in Base62 (6ˆ62) ~ 1.7*10ˆ48

But most definitely it's faster and produces more readable hashes (without `-` and `_`)

**Review by martindisch — APPROVED** [review]
I like it. Still want to incorporate some benchmarks in the future though, would be cool to see these differences.

File: swc/src/hash.rs [hunk 1]
```diff
@@ -1,41 +1,31 @@
...
 }
 
 #[cfg(test)]
 mod tests {
     use super::*;
 
     #[test]
-    fn hey() {
-        assert_eq!(hash("hey", 5), "XldIH");
+    fn shor() {
```
Comment (martindisch) [inline swc/src/hash.rs:15]: One small nitpick, unless it's a subtle joke on shortening the word short ;-)

```suggestion
    fn short() {
```

**Comment by martindisch** [comment]
Fun fact: the new hash function is also about 13x faster (admittedly that doesn't matter at all in the context of what the plugin does, but cool beans anyway) and more importantly it makes the final Wasm binary 143 KB lighter.

I've been trying to figure out how we can accurately benchmark the plugin not only as native code, but also while running in a Wasm runtime, which is how it will be executed by SWC. I'm merging this one now to prevent any conflicts with this other work, because that will require some shuffling around of stuff.

**Pull request merged**